Spring Boot examples not working: Process variables are not provided to Task

I tried to run the Test in both `activiti-api-basic-full-example-nobean` and `activiti-api-basic-full-example-bean` and I found at least 3 issues:

1) The test doesn't contain any Assertions so it always passes even if in reality it shouldn't
2) Half of the times this row returns an empty list of Tasks so nothing happens (maybe the scheduler should wait longer?):
` Page<Task> tasks = taskRuntime.tasks(Pageable.of(0, 10));`

3) This is the actual problem: if there is a Task available to process, the application throws because the List of variables is EMPTY. But the test passes because there are no assertions.
```
java.lang.IndexOutOfBoundsException: Index 0 out of bounds for length 0
	at java.base/jdk.internal.util.Preconditions.outOfBounds(Preconditions.java:64) ~[na:na]
	at java.base/jdk.internal.util.Preconditions.outOfBoundsCheckIndex(Preconditions.java:70) ~[na:na]
	at java.base/jdk.internal.util.Preconditions.checkIndex(Preconditions.java:248) ~[na:na]
	at java.base/java.util.Objects.checkIndex(Objects.java:372) ~[na:na]
	at java.base/java.util.ArrayList.get(ArrayList.java:458) ~[na:na]
	at org.activiti.examples.DemoApplication.checkAndWorkOnTasksWhenAvailable(DemoApplication.java:126) ~[classes/:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[na:na]
	at java.base/jdk.internal.reflect.NativeMethodAccessorImpl.invoke(NativeMethodAccessorImpl.java:62) ~[na:na]
	at java.base/jdk.internal.reflect.DelegatingMethodAccessorImpl.invoke(DelegatingMethodAccessorImpl.java:43) ~[na:na]
	at java.base/java.lang.reflect.Method.invoke(Method.java:566) ~[na:na]
	at org.springframework.scheduling.support.ScheduledMethodRunnable.run(ScheduledMethodRunnable.java:84) ~[spring-context-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at org.springframework.scheduling.support.DelegatingErrorHandlingRunnable.run(DelegatingErrorHandlingRunnable.java:54) ~[spring-context-5.2.7.RELEASE.jar:5.2.7.RELEASE]
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:515) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset$$$capture(FutureTask.java:305) ~[na:na]
	at java.base/java.util.concurrent.FutureTask.runAndReset(FutureTask.java) ~[na:na]
	at java.base/java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:305) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128) ~[na:na]
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628) ~[na:na]
	at java.base/java.lang.Thread.run(Thread.java:834) ~[na:na]
```

This is an issue I experienced in my own project after upgrading Activiti. I know that it was working with version 7.1.123.
***In my application, the `integrationContext` that is provided to my `Connector` beans does not contain any variables.***

I also tried the activiti-api-spring-integration-example and same problem: 
```
  @Bean
    public Connector processTextConnector() {
        return integrationContext -> {
            Map<String, Object> inBoundVariables = integrationContext.getInBoundVariables();
```
This returns an empty map and then obviously there is a NPE:
```
Caused by: java.lang.NullPointerException
	at org.activiti.examples.DemoApplication.lambda$processTextConnector$0(DemoApplication.java:132)
	at org.activiti.runtime.api.connector.DefaultServiceTaskBehavior.execute(DefaultServiceTaskBehavior.java:47)
	at org.activiti.engine.impl.delegate.ActivityBehaviorInvocation.invoke(ActivityBehaviorInvocation.java:39)
	at org.activiti.engine.impl.delegate.invocation.DelegateInvocation.proceed(DelegateInvocation.java:39)
	at org.activiti.engine.impl.delegate.invocation.DefaultDelegateInterceptor.handleInvocation(DefaultDelegateInterceptor.java:29)
	at org.activiti.engine.impl.bpmn.behavior.ServiceTaskDelegateExpressionActivityBehavior.execute(ServiceTaskDelegateExpressionActivityBehavior.java:92)
	... 65 more
```

--- a/activiti-examples/activiti-api-basic-full-example-bean/src/main/resources/processes/categorize-human-content-extensions.json
+++ b/activiti-examples/activiti-api-basic-full-example-bean/src/main/resources/processes/categorize-human-content-extensions.json
@@ -1,0 +1,34 @@
+{
+  "id":"categorizeHumanProcessModel",
+  "type": "PROCESS",
+  "name": "process",
+  "extensions": {
+    "categorizeHumanProcess": {
+      "properties": {
+        "557dfb49-456b-492d-b99c-0a49ee325e77": {
+          "id": "557dfb49-456b-492d-b99c-0a49ee325e77",
+          "name": "content",
+          "type": "object",
+          "required": true
+        }
+      },
+      "mappings": {
+        "Task_1ylvdew": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_0snvh02": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_1asxw87": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        }
+      }
+    }
+  }
+}

--- a/activiti-examples/activiti-api-basic-full-example-nobean/src/main/resources/processes/categorize-human-content-extensions.json
+++ b/activiti-examples/activiti-api-basic-full-example-nobean/src/main/resources/processes/categorize-human-content-extensions.json
@@ -1,0 +1,34 @@
+{
+  "id":"categorizeHumanProcessModel",
+  "type": "PROCESS",
+  "name": "process",
+  "extensions": {
+    "categorizeHumanProcess": {
+      "properties": {
+        "557dfb49-456b-492d-b99c-0a49ee325e77": {
+          "id": "557dfb49-456b-492d-b99c-0a49ee325e77",
+          "name": "content",
+          "type": "object",
+          "required": true
+        }
+      },
+      "mappings": {
+        "Task_1ylvdew": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_0snvh02": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_1asxw87": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        }
+      }
+    }
+  }
+}

--- a/activiti-examples/activiti-api-basic-process-example/src/main/resources/processes/categorize-content-extensions.json
+++ b/activiti-examples/activiti-api-basic-process-example/src/main/resources/processes/categorize-content-extensions.json
@@ -1,0 +1,34 @@
+{
+  "id":"categorizeProcessModel",
+  "type": "PROCESS",
+  "name": "process",
+  "extensions": {
+    "categorizeProcess": {
+      "properties": {
+        "557dfb49-456b-492d-b99c-0a49ee325e77": {
+          "id": "557dfb49-456b-492d-b99c-0a49ee325e77",
+          "name": "content",
+          "type": "string",
+          "required": true
+        }
+      },
+      "mappings": {
+        "Task_1ylvdew": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_0snvh02": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_1asxw87": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        }
+      }
+    }
+  }
+}

--- a/activiti-examples/activiti-api-spring-integration-example/src/main/resources/processes/categorize-text-extensions.json
+++ b/activiti-examples/activiti-api-spring-integration-example/src/main/resources/processes/categorize-text-extensions.json
@@ -1,0 +1,34 @@
+{
+  "id":"categorizeProcessModel",
+  "type": "PROCESS",
+  "name": "process",
+  "extensions": {
+    "categorizeProcess": {
+      "properties": {
+        "557dfb49-456b-492d-b99c-0a49ee325e77": {
+          "id": "557dfb49-456b-492d-b99c-0a49ee325e77",
+          "name": "content",
+          "type": "string",
+          "required": true
+        }
+      },
+      "mappings": {
+        "Task_1ylvdew": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_0snvh02": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_1asxw87": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        }
+      }
+    }
+  }
+}

--- a/activiti-examples/activiti-api-web-example/src/main/resources/processes/categorize-text-extensions.json
+++ b/activiti-examples/activiti-api-web-example/src/main/resources/processes/categorize-text-extensions.json
@@ -1,0 +1,34 @@
+{
+  "id":"categorizeProcessModel",
+  "type": "PROCESS",
+  "name": "process",
+  "extensions": {
+    "categorizeProcess": {
+      "properties": {
+        "557dfb49-456b-492d-b99c-0a49ee325e77": {
+          "id": "557dfb49-456b-492d-b99c-0a49ee325e77",
+          "name": "fileContent",
+          "type": "string",
+          "required": true
+        }
+      },
+      "mappings": {
+        "Task_1ylvdew": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_0snvh02": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        },
+        "Task_1asxw87": {
+          "mappingType": "MAP_ALL",
+          "inputs": {},
+          "outputs": {}
+        }
+      }
+    }
+  }
+}

